src/exe_reader.cpp

```diff
@@ -34,19 +34,31 @@ namespace {
 EXEReader::EXEReader(Filesystem_Stream::InputStream core) : corefile(std::move(core)) {
 	// The Incredibly Dumb PE parser (tm)
 	// Extracts data from the resource section for engine detection and can read ExFont.
-	uint32_t ofs = GetU32(0x3C);
+	uint32_t ofs = GetU32(0x3C); // PE header offset
 	uint16_t machine = GetU16(ofs + 4);
-	if (machine != 0x14c) {
+	if (machine != 0x14c && machine != 0x8664) {
 		// FIXME: Newer versions of Maniac Patch can be 64 Bit
-		Output::Debug("EXEReader: Machine type is not i386 ({:#x})", machine);
-		file_info.is_i386 = false;
+		Output::Debug("EXEReader: Machine type is not i386 or amd64 ({:#x})", machine);
+		file_info.machine_type = MachineType::Unknown;
 		return;
 	}
 
+	if (machine == 0x14c) {
+		file_info.machine_type = MachineType::i386;
+	} else {
+		file_info.machine_type = MachineType::amd64;
+	}
+
```

Comment:
I think this block could be simplified as a switch/case or just reorder to have the "unsupported" else case last.
Also the comment about 64 bit maniac patch seems resolved now?

Reply:
yeah that code was bad. Is now a switch-case with error in default.

Added a similar check to the PE/PE+ header magic.